compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -710,6 +741,26 @@ private BIRVariableDcl getSelf(BSymbol receiver) {
         return self;
     }
 
+    private void analyzeParametersAndReturnType(BLangFunction astFunc) {
+        BLangType typeNode = astFunc.returnTypeNode;
+        if (typeNode != null) {
+            typeNode.accept(this);
+        }
+        for (BLangSimpleVariable parameter : astFunc.requiredParams) {
+            typeNode = parameter.typeNode;
+            if (typeNode != null) {
+                typeNode.accept(this);
+            }
+        }
+        BLangSimpleVariable restParam = astFunc.restParam;
+        if (restParam != null) {
+            typeNode = restParam.typeNode;
+            if (typeNode != null) {
+                typeNode.accept(this);
+            }
+        }
```

Comment:
I would visit these in order. Required params, rest param, return type.

Reply:
Updated